crates/ln-dlc-node/src/ln_dlc_wallet.rs

```diff
@@ -277,53 +277,57 @@ impl<S: TenTenOneStorage, N: Storage> dlc_manager::Wallet for LnDlcWallet<S, N>
         _: Option<u64>,
         lock_utxos: bool,
     ) -> Result<Vec<Utxo>, Error> {
```

Comment:
🤔 It looks to me we might run into a risk of race conditions here, in case two threads work in this function in parallel.

The reading and writing of the `reserved_outpoints` should happen mutually exclusive from any other thread.

Reply:
Fair point. I move this logic into the wallet